docs(swarm local address): add documentation spec for swarm.localAddrs (#477)
ipfs/interface-js-ipfs-core: SPEC/SWARM.md

# Swarm API

* [swarm.addrs](#swarmaddrs)

> * [swarm.localAddrs](#swarmlocaladdrs)

Reviewer: Please can this be arranged alphabetically? i.e. `swarm.localAddrs` should be after `swarm.disconnect` and before `swarm.peers`.

Author: Items in this list are already ordered logically than alphabetically. Should I fix them up too?

Reviewer: Yes please

A great source of [examples][] can be found in the tests for this API.

#### `swarm.localAddrs`

> > List of known local addresses of each peer connected.

Reviewer: Not quite correct, these are the addresses your node is listening on.

```suggestion
> List local addresses this node is listening on.
```

Author: Fixed.

##### `ipfs.swarm.localAddrs([callback])`

> `callback` must follow `function (err, peerInfos) {}` signature, where `err` is an error if the operation was not successful. `peerInfos` will be an array of [`PeerInfo`](https://github.com/libp2p/js-peer-info)s.

Reviewer: It actually returns [`Multiaddr`](https://github.com/multiformats/js-multiaddr/) instances _not_ `PeerInfo` instances.

Author: Fixed.
